telegram/src/test/kotlin/com/github/kotlintelegrambot/updater/BoundLooper.kt

```diff
@@ -1,15 +1,22 @@
 package com.github.kotlintelegrambot.updater
 
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
 /**
  * Looper implementation for testing purposes. It will loop [loopIterations] times and then stop.
  */
-class BoundLooper : Looper {
+class BoundLooper(coroutineDispatcher: CoroutineDispatcher) : Looper {
 
+    private val scope: CoroutineScope = CoroutineScope(coroutineDispatcher)
     var loopIterations = 0
 
-    override fun loop(loopBody: () -> Unit) {
-        repeat(loopIterations) {
-            loopBody()
+    override fun loop(loopBody: suspend () -> Unit) {
+        scope.launch {
```

Comment:
Is there a reason why we need to launch a coroutine here for some of the tests or could we keep this being executed in the caller context?

Reply:
Ah true, that makes sense. Thanks for the explanation.